qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -585,6 +588,82 @@ def _check_special_columns(cls, md_template, obj):
             cls._check_template_special_columns(md_template, obj))
 
     @classmethod
+    def _clean_validate_template(cls, md_template, study_id, obj,
```

Comment:
That's a good point. If the template being tested is a prep template, the process will fail later because the samples are not found in the sample template. If it is the study_id, there is no way, since the object still doesn't exist. But 2 things: (1) this was already like that in the previous implementation; and (2) this is a private function that should not be called from outside the class, so if somebody is messing up with this function they should know what they're doing (and this is completely pythonic).